src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -1605,10 +1605,11 @@ void UpdateMemoryCost(in UInt256 position, in UInt256 length)
                         }
 
                         stack.PopUInt256(out UInt256 memPosition);
+                        
                         Span<byte> data = stack.PopBytes();
                         UpdateMemoryCost(in memPosition, 32);
                         vmState.Memory.SaveWord(in memPosition, data);
-                        if (_txTracer.IsTracingInstructions) _txTracer.ReportMemoryChange((long) memPosition, data.PadLeft(32));
+                        if (_txTracer.IsTracingInstructions) _txTracer.ReportMemoryChange((long) memPosition, data.SliceWithZeroPadding(0, 32, PadDirection.Left));
```

Comment:
This seems very risky - we are passing a Span pointing at the internal stack structure to an an externally controlled interface implementation (possibly coming from plugins.

Reply:
It has default implementation, so if someone implements it he better knows what he is doing :)